grails-app/taglib/transmart/UserProfileTagLib.groovy

```diff
@@ -0,0 +1,16 @@
+package transmart
+
+class UserProfileTagLib {
+
+    def userProfileService
+
+    def userProfile = { attrs ->
+        if (userProfileService) {
+            out << "<li><a href=\"" + g.createLink(controller:"userProfile", action:"index") + "\">User profile</a></li>"
```

Comment:
you should omit `g.` when calling taglib methods in taglibs and controllers. also omit `action:"index"` since it's the default action